nodebuilder/share/module.go

```diff
@@ -185,6 +110,46 @@ func shrexGetterComponents(cfg *Config) fx.Option {
 			}),
 		)),
 	)
+
+	switch tp {
+	case node.Light:
+		return fx.Options(
+			opts,
+			// shrexsub broadcaster stub for daser
+			fx.Provide(func() shrexsub.BroadcastFn {
+				return func(context.Context, shrexsub.Notification) error {
+					return nil
+				}
+			}),
+		)
+	case node.Full:
+		return fx.Options(
+			opts,
+			shrexServerComponents(cfg),
+			fx.Provide(getters.NewStoreGetter),
+			fx.Provide(func(shrexSub *shrexsub.PubSub) shrexsub.BroadcastFn {
+				return shrexSub.Broadcast
+			}),
+		)
+	case node.Bridge:
+		return fx.Options(
+			opts,
+			shrexServerComponents(cfg),
+			fx.Provide(getters.NewStoreGetter),
+			fx.Provide(func(shrexSub *shrexsub.PubSub) shrexsub.BroadcastFn {
+				return shrexSub.Broadcast
+			}),
+			fx.Invoke(func(lc fx.Lifecycle, sub *shrexsub.PubSub) error {
+				lc.Append(fx.Hook{
+					OnStart: sub.Start,
+					OnStop:  sub.Stop,
+				})
+				return nil
+			}),
```

Comment:
shrexSub subscriber should be part of Full and Light and not be part of Bridge. Somehow it is the opposite in this constructor

Reply:
@walldiss shrexsub subscriber lifecycle is unfortunately implicitly managed inside peer manager, but we do not pass shrexsub to peer manager for bridge nodes bc we do not care to listen to shrexsub pools. BN still needs to broadcast to shrexsub though so we need to manage shrexsub lifecycle here.

Reply:
https://github.com/celestiaorg/celestia-node/pull/3188/commits/da20efc103bf02797140b067d0913396fc196089

this unlocked new change though

Reply:
Ughhh. Good to know. Who might have though it is good idea to manage lifecycle in peer-manager 🥇 https://github.com/celestiaorg/celestia-node/issues/3433
I'm fine if it will be fixed later.